### PR TITLE
Improved error handling

### PR DESCRIPTION
**Pull request by davidfurey**
## Why are you doing this?

To provide immediate feedback to the customer so they know to try again now or later.

Also improves exception logging generally

### CODE REVIEW

File: common/src/main/scala/com/gu/zuora/model/response/Responses.scala [hunk 1]
```diff
@@ -14,12 +18,24 @@ object ZuoraError {
 
 case class ZuoraError(Code: String, Message: String)
 
-object ZuoraErrorResponse {
+object ZuoraErrorResponse extends LazyLogging {
```
Comment (rupertbates) [inline common/src/main/scala/com/gu/zuora/model/response/Responses.scala]: Don't need to extend LazyLogging here I don't think

File: common/src/main/scala/com/gu/zuora/model/response/Responses.scala [hunk 2]
```diff
@@ -14,12 +18,24 @@ object ZuoraError {
...
+    } else {
+      None
+    }
+  }
 }
 
-case class ZuoraErrorResponse(success: Boolean, errors: List[ZuoraError]) extends Throwable with ZuoraResponse {
-  def asRetryException: RetryException = new RetryNone(cause = this)
+case class ZuoraErrorResponse(success: Boolean, errors: List[ZuoraError])
+    extends Throwable(errors.asJson.noSpaces) with ZuoraResponse {
```
Comment (rupertbates) [inline common/src/main/scala/com/gu/zuora/model/response/Responses.scala]: spaces2 might be easier to read when debugging than noSpaces

File: common/src/main/scala/com/gu/zuora/model/response/Responses.scala (hunk 1 above)
Comment (davidfurey) [inline common/src/main/scala/com/gu/zuora/model/response/Responses.scala]: Thanks, I forgot to remove that after some debugging.

File: common/src/main/scala/com/gu/zuora/model/response/Responses.scala (hunk 2 above)
Comment (davidfurey) [inline common/src/main/scala/com/gu/zuora/model/response/Responses.scala]: Good idea

**Pull request merged**